api/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM golang:1.17
+
+RUN go version
+ENV GOPATH=/
+
+# Set destination for COPY
+WORKDIR /api
+
+COPY ./ ./
```

Comment:
Docker can cache steps for the next builds. E.g. if your source code didn't change and you run build for the second time it will take image layers (each line in dockerfile creates new layer) from cache. To make this caching better you should be careful with order of your docker commands, they should come from those to change less frequently to those that change more frequently. First usually come installation of OS packages, second copy and install of app libraries, third copy of the source code.
E.g. here is the example from docker docs https://docs.docker.com/language/golang/build-images/
![image](https://user-images.githubusercontent.com/84133701/144626103-ea435540-0c72-4ba3-ba9d-238d15f3ce3e.png)
As you see they first download modules and secondly copy source code.